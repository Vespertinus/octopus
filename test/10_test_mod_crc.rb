#!/usr/bin/ruby1.9.1

$:.push 'test/lib'
require 'standalone_env'

class Env < StandAloneEnv
  def config
    super + <<EOD
rows_per_wal=32
run_crc_delay=0.1
object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
EOD
  end
end

env = Env.clean

def wait_for(n=100)
  n.times do
    return if yield
    sleep 0.05
  end
  raise "wait_for failed"
end

env.with_server do
  16.times {|i| insert [i,i]}
  Process.kill('USR1', env.pid)
  32.times {|i| insert [i, i + 1]}
  wait_for { lx = open('|./octopus --cat 00000000000000000032.xlog').lines.grep(/run_crc/).length > 0 }
end

env.with_server do
  puts `./octopus --cat 00000000000000000032.xlog | grep run_crc | sed 's/ tm:[^ ]*//'`
  puts File.open('octopus.log').grep(/mismatch/)

  16.times {|i| insert [i,i]}
  Process.kill('USR1', env.pid)
  wait_for { File.exist? '00000000000000000066.snap' }
end

env.with_server do
  32.times {|i| insert [i, i + 1]}
  wait_for { open('|./octopus --cat 00000000000000000096.xlog').lines.grep(/run_crc/).length > 0 }
end

env.with_server do
  puts `./octopus --cat 00000000000000000096.xlog | grep run_crc | sed 's/ tm:[^ ]*//'`
  puts File.open('octopus.log').grep(/mismatch/)
  puts `grep -P '>E|mismatch' octopus.log*`
end
