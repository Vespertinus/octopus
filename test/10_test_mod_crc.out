# box.insert([0, 0])
1

# box.insert([1, 1])
1

# box.insert([2, 2])
1

# box.insert([3, 3])
1

# box.insert([4, 4])
1

# box.insert([5, 5])
1

# box.insert([6, 6])
1

# box.insert([7, 7])
1

# box.insert([8, 8])
1

# box.insert([9, 9])
1

# box.insert([10, 10])
1

# box.insert([11, 11])
1

# box.insert([12, 12])
1

# box.insert([13, 13])
1

# box.insert([14, 14])
1

# box.insert([15, 15])
1

# box.insert([0, 1])
2

# box.insert([1, 2])
2

# box.insert([2, 3])
2

# box.insert([3, 4])
2

# box.insert([4, 5])
2

# box.insert([5, 6])
2

# box.insert([6, 7])
2

# box.insert([7, 8])
2

# box.insert([8, 9])
2

# box.insert([9, 10])
2

# box.insert([10, 11])
2

# box.insert([11, 12])
2

# box.insert([12, 13])
2

# box.insert([13, 14])
2

# box.insert([14, 15])
2

# box.insert([15, 16])
2

# box.insert([16, 17])
1

# box.insert([17, 18])
1

# box.insert([18, 19])
1

# box.insert([19, 20])
1

# box.insert([20, 21])
1

# box.insert([21, 22])
1

# box.insert([22, 23])
1

# box.insert([23, 24])
1

# box.insert([24, 25])
1

# box.insert([25, 26])
1

# box.insert([26, 27])
1

# box.insert([27, 28])
1

# box.insert([28, 29])
1

# box.insert([29, 30])
1

# box.insert([30, 31])
1

# box.insert([31, 32])
1

lsn:50 scn:50 t:run_crc 0.0.0.0:0 SCN:49 log:0xb2e3af90 mod:0x00000000
# box.insert([0, 0])
2

# box.insert([1, 1])
2

# box.insert([2, 2])
2

# box.insert([3, 3])
2

# box.insert([4, 4])
2

# box.insert([5, 5])
2

# box.insert([6, 6])
2

# box.insert([7, 7])
2

# box.insert([8, 8])
2

# box.insert([9, 9])
2

# box.insert([10, 10])
2

# box.insert([11, 11])
2

# box.insert([12, 12])
2

# box.insert([13, 13])
2

# box.insert([14, 14])
2

# box.insert([15, 15])
2

# box.insert([0, 1])
2

# box.insert([1, 2])
2

# box.insert([2, 3])
2

# box.insert([3, 4])
2

# box.insert([4, 5])
2

# box.insert([5, 6])
2

# box.insert([6, 7])
2

# box.insert([7, 8])
2

# box.insert([8, 9])
2

# box.insert([9, 10])
2

# box.insert([10, 11])
2

# box.insert([11, 12])
2

# box.insert([12, 13])
2

# box.insert([13, 14])
2

# box.insert([14, 15])
2

# box.insert([15, 16])
2

# box.insert([16, 17])
2

# box.insert([17, 18])
2

# box.insert([18, 19])
2

# box.insert([19, 20])
2

# box.insert([20, 21])
2

# box.insert([21, 22])
2

# box.insert([22, 23])
2

# box.insert([23, 24])
2

# box.insert([24, 25])
2

# box.insert([25, 26])
2

# box.insert([26, 27])
2

# box.insert([27, 28])
2

# box.insert([28, 29])
2

# box.insert([29, 30])
2

# box.insert([30, 31])
2

# box.insert([31, 32])
2

lsn:99 scn:99 t:run_crc 0.0.0.0:0 SCN:98 log:0x3390e58a mod:0x00000000

